Sources/ManifestSerializer/Manifest+parse.swift

```diff
@@ -66,17 +67,17 @@ private func parse(path manifestPath: String, swiftc: String, libdir: String) th
 #endif
     cmd += [manifestPath]
 
-    //Create and open a temporary file to write toml to
-    let filePath = Path.join(manifestPath.parentDirectory, ".Package.toml")
-    let fp = try fopen(filePath, mode: .Write)
-    defer { fp.closeFile() }
-
-    //Pass the fd in arguments
-    cmd += ["-fileno", "\(fp.fileDescriptor)"]
-    try system(cmd)
-
-    let toml = try fopen(filePath).reduce("") { $0 + "\n" + $1 }
-    try unlink(filePath) //Delete the temp file after reading it
+    //Previous implementation used a temporary file, however NSTask doesn't
+    //allow child process to access parent's file descriptors. So instruct
+    //swiftc to dump the package information on stdout.
+    #if os(OSX) || os(iOS)
+        let handle = NSFileHandle.standardOutput()
+    #else
+        let handle = NSFileHandle.fileHandleWithStandardOutput()
+    #endif
+
+    cmd += ["-fileno",  String(handle.fileDescriptor)]
+    let toml = try Utility.popen(cmd)
```

Comment:
It just doesn't seem worth it to me. I think this is a case where it makes the most sense to use posix_spawn directly rather than work harder just to use `NSTask`.